src/main/java/mate/academy/spring/service/impl/CustomUserDetailsService.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.spring.service.impl;
+
+import static org.springframework.security.core.userdetails.User.UserBuilder;
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findByEmail(username).orElseThrow(() ->
+                new UsernameNotFoundException("User with email " + username + "not found"));
+
+        UserBuilder userBuilder = withUsername(username).password(user.getPassword())
+                .roles(user.getRoles().stream().map(
+                        r -> r.getRoleName().name()).toArray(String[]::new));
```

Comment:
do not use the short name
```suggestion
                        role -> role.getRoleName().name()).toArray(String[]::new));
```